pkg/controller/jobs/jobset/jobset_webhook.go

```diff
@@ -57,9 +67,33 @@ func (w *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log.V(5).Info("Applying defaults", "jobset", klog.KObj(jobSet))
 
 	jobframework.ApplyDefaultForSuspend(jobSet, w.manageJobsWithoutQueueName)
+
+	if canDefaultManagedBy(jobSet.Spec.ManagedBy) {
+		localQueueName, found := jobSet.Labels[constants.QueueLabel]
+		if !found {
+			return nil
+		}
+		clusterQueueName, err := w.queues.ClusterQueueFromLocalQueue(queue.QueueKey(jobSet.ObjectMeta.Namespace, localQueueName))
+		if err != nil {
+			return err
+		}
+		for _, admissionCheck := range w.cache.AdmissionChecksForClusterQueue(clusterQueueName) {
+			if admissionCheck.Controller == multikueue.ControllerName {
+				log.V(5).Info("Defaulting ManagedBy", "jobset", klog.KObj(jobSet), "oldManagedBy", jobSet.Spec.ManagedBy, "managedBy", multikueue.ControllerName)
+				jobSet.Spec.ManagedBy = ptr.To(multikueue.ControllerName)
+				break
+			}
+		}
+	}
+
 	return nil
 }
 
+func canDefaultManagedBy(jobSetSpecManagedBy *string) bool {
+	return features.Enabled(features.MultiKueue) &&
+		(jobSetSpecManagedBy == nil || *jobSetSpecManagedBy == jobsetapi.JobSetControllerName)
```

Comment:
```suggestion
	return features.Enabled(features.MultiKueue) && ptr.Equal(jobSetSpecManagedBy, jobsetapi.JobSetControllerName)
```

Reply:
Will this work? If `jobSet.spec.managedBy==nil`, then we want to default.

Reply:
Oh, I see.
You're right.